oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/SegmentAnalysisListener.java

```diff
@@ -153,6 +142,14 @@ public void parseSegment(SegmentObject segmentObject) {
         });
         final long accurateDuration = endTimestamp - startTimestamp;
         duration = accurateDuration > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) accurateDuration;
+
+        if (sampleStatus.equals(SAMPLE_STATUS.UNKNOWN) || sampleStatus.equals(SAMPLE_STATUS.IGNORE)) {
+            if (sampler.shouldSample(segmentObject.getTraceId()) || forceSaveErrorSegment) {
```

Comment:
This will be always true in default, as `forceSaveErrorSegment == true`. I think you mischanged this?

Reply:
You should `isError && forceSaveErrorSegment` from my understanding, isn't it?

Reply:
my bad, changed.